src/com/twitter/intellij/pants/execution/PantsMakeBeforeRun.java

```diff
@@ -348,9 +329,124 @@ else if (output.contains("[error]")) {
+
+            OpenFileHyperlinkInfo linkInfo = new OpenFileHyperlinkInfo(
+              project,
+              result.get().getVirtualFile().get(),
+              result.get().lineNumber - 1, // line number needs to be 0 indexed
+              result.get().columnNumber - 1 // column number needs to be 0 indexed
+            );
+            int startHyperlink = entireLength - line.length() + line.indexOf(ERROR_TAG);
+
+            return new Result(
+              startHyperlink,
+              entireLength,
+              linkInfo,
+              null // TextAttributes, going with default hence null
+            );
+          }
+        }
+        return null;
+      }
+    };
+
+    ApplicationManager.getApplication().invokeLater(new Runnable() {
+      @Override
+      public void run() {
+        executionConsole.addMessageFilter(filter);
+        executionConsole.print(message, type);
+      }
+    }, ModalityState.NON_MODAL);
+  }
+
+  /**
+   * Encapsulate the result of parsed data.
+   */
+  static class ParseResult {
+    String filePath;
+    int lineNumber;
+    int columnNumber;
+
+    ParseResult(String filePath, int lineNumber, int columnNumber) {
+      this.filePath = filePath;
+      this.lineNumber = lineNumber;
+      this.columnNumber = columnNumber;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      if (obj == null) {
+        return false;
+      }
+      if (getClass() != obj.getClass()) {
+        return false;
+      }
+      ParseResult other = (ParseResult) obj;
+      return Objects.equals(filePath, other.filePath)
+             && Objects.equals(lineNumber, other.lineNumber)
+             && Objects.equals(columnNumber, other.columnNumber);
+    }
+
+    public String getFilePath() {
+      return filePath;
+    }
+
+    public Optional<VirtualFile> getVirtualFile() {
+      return Optional.ofNullable(VirtualFileManager.getInstance().findFileByUrl("file://" + filePath));
+    }
+
+    public int getLineNumber() {
+      return lineNumber;
+    }
+
+    public int getColumnNumber() {
+      return columnNumber;
+    }
+  }
+
+  /**
+   * This function parses Pants output against known file extension and tag,
+   * and returns (Path to mentioned file, line number, column number)
+   * encapsulated in `ParseResult` object.
+   *
+   * @param line original Pants output
+   * @param ext  known file extension. e.g. ".java", ".scala"
+   * @param tag  known tag. e.g. [error]
+   * @return `ParseResult` instance
+   */
+  public static Optional<ParseResult> parseErrorLocation(String line, String ext, String tag) {
+    if (!line.contains(ext) || !line.contains(tag)) {
+      return Optional.empty();
+    }
+
+    String[] splitByColon = line.split(":");
+    if (splitByColon.length < 3) {
+      return Optional.empty();
+    }
+
+    try {
+      // filePath path is between tag and first colon
+      String filePath = splitByColon[0].substring(splitByColon[0].indexOf(tag) + tag.length()).trim();
```

Comment:
It's okay for `String filePath` to be `Assert.assertEquals("0` for example, because [the hyperlink construction will check if it is an actual file](https://github.com/pantsbuild/intellij-pants-plugin/pull/242/files/c3d1080f4acc15027ff7b6d3b505d8ee7c693dc5#diff-77d87e704dfb9b6b30ffaa3a11f7d6cdR343). If not, it will there will be no hyperlinks. 

I will add more doc to this to be more clear.